src/ucp/core/ucp_ep.c

```diff
@@ -1838,6 +1838,40 @@ ucp_lane_index_t ucp_ep_lookup_lane(ucp_ep_h ucp_ep, uct_ep_h uct_ep)
     return UCP_NULL_LANE;
 }
 
+ucp_lane_index_t ucp_ep_find_wireup_ep_lane(ucp_ep_h ep)
+{
+    ucp_lane_index_t lane;
+
+    for (lane = 0; lane < ucp_ep_num_lanes(ep); lane++) {
+        if (ucp_wireup_ep_test(ucp_ep_get_lane(ep, lane))) {
+            return lane;
+        }
+    }
```

Comment:
Wouldn't it be better to save `wireup_ep` in worker, and then access it in O(1)?